TicTacToe.cpp

```diff
@@ -0,0 +1,250 @@
+
+bool checksplit(int split,std::vector<std::string> &feld);
+
+bool checkMainDiagonal(std::vector<std::string> &feld);
+
+bool checkSecondaryDiagonal(std::vector<std::string> &feld);
+
+
+int main(int argc, char const *argv[])
+{
+   // Datentypen
+   // 2d Array/Vector
+   std::vector<std::string> feld = {
+      "123",
+      "456",
+      "789"
+   };
+   // Spieler mit zugehörigen Zeichen
+   char player1 = 'X'; // 'X' bzw. 'O'
+   char player2 = 'O';
+   int eingabe = 0;
+
+   while (!gameover(feld))
+   {
+      if (zuege % 2 != 0)
+      {
+         std::cout << "Spieler " << player1 << " ist am Zug\n";
+         std::cout << "\n";
+         print_board(feld);
+         std::cout << "\n";
+         std::cout << "Bitte Feldnummer eingeben\n";
+         std::cin >> eingabe;
+         std::cout <<"\n";
+         insert_move(eingabe, player1, feld);
+      } 
+      else 
+      {
+         std::cout << "Spieler " << player2 << " ist am Zug\n";
+         std::cout << "\n";
+         print_board(feld);
+         std::cout << "\n";
+         std::cout << "Bitte Feldnummer eingeben\n";
+         std::cin >> eingabe;
+         std::cout <<"\n";
+         insert_move(eingabe, player2, feld);
+      }
+      zuege++;
+   }
+   print_board(feld);
+   //Gewinner ausgeben
+   bool unentschieden = false;
+   char ausgabe = findTheWinner(feld, unentschieden);
+   if (!unentschieden){
+      std::cout << ausgabe << " ist der Gewinner.";
+   }
+   else{
+      std::cout << "Das Spiel ist unentschieden ausgegangen.";
+   }   
+   return 0;
+}
+
+bool gameover(std::vector<std::string> &feld) {
+   if (zuege > 9)
+   {
+      return true;
+   }
+   else
+   {
+      bool var = false;
+      // Überprüfen der einzelnen Zeilen
+      // Überprüfen Zeile 1
+      for (int i = 0; i < 3; i++) //Zeilen überprüfen
+      {
+         var = checkline(i, feld);
+         if (var)
+         {
+            return true;
+         }
+      }
+      for (int j = 0; j < 3; j++) //splitn überprüfen
+      {
+         var = checksplit(j, feld); 
+         if (var)
+         {
+            return true;
+         }
+      }
+      var = checkMainDiagonal(feld); // 1. Diagonale wird überprüft
+      if (var)
+      {
+         return true;
+      }
+      var = checkSecondaryDiagonal(feld); // 2. Diagonale wird überprüft
+      if (var)
+      {
+         return true;
+      }
+   }
+}
+
+//findTheWinner: Funktion, die Gewinner ermittelt und zurückgibt
+
+char findTheWinner(std::vector<std::string> &feld, bool &unentschieden) {
+   if (zuege <= 9)
+   {
+      bool var = false;
+      // Überprüfen der einzelnen Zeilen
+      // Überprüfen Zeile 1
+      for (int i = 0; i < 3; i++) //Zeilen überprüfen
+      {
+         var = checkline(i, feld);
+         if (var)
+         {
+            return feld[i][0];
+         }
+      }
+      for (int j = 0; j < 3; j++) //splitn überprüfen
+      {
+         var = checksplit(j, feld);
+         if (var)
```

Comment:
direkt ins if statement, die Variable `var` wird hier nicht unbedingt gebraucht.
Allgemein bei den if statements.

```suggestion
         if (checksplit(j, feld))
```